source/dagger/gameplay/platformer/platformer_controller.cpp

```diff
@@ -27,18 +27,50 @@ void PlatformerControllerSystem::SpinUp()
 void PlatformerControllerSystem::Run()
 {
     Engine::Registry().view<InputReceiver, Sprite, Animator, PlatformerCharacter>().each(
-        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, const PlatformerCharacter& char_)
+        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, PlatformerCharacter& char_)
         {
+            Float32 roll = input_.values.at("roll");
             Float32 run = input_.values.at("run");
-            if (run == 0)
+            
+            if (roll == 1)
             {
-                AnimatorPlay(animator_, "souls_like_knight_character:IDLE");
+                char_.isRolling = true;
+                AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
+
+                // Ex: If the player is pressing the left movement key and roll Key
+                // allow him to do that. Alligning the sprite_.scale.x with run variable enables
+                // the character to start rolling in the direction it is facing
+                if (run != 0)
+                {
+                    sprite_.scale.x = run;
+                }
+
+                if (char_.isRolling)
+                {
+                    if (char_.timeRolling < char_.rollingTime)
```

Comment:
so it's just rolling for 5 frames? why not to make it Float32 time dependent? 0.5 sec or something?